### ISSUE TITLE
Langversion error while building from Rider

### ISSUE THREAD

**Issue by RDeluxe** [JetBrains/resharper-unity#472]
Hi,

This may very well be a stupid question, but I'm getting `Error Invalid option '6' for /langversion; must be ISO-1, ISO-2, 3, 4, 5 or Default` errors while compiling from Rider.

Unity : 2017.3.1.f1
Rider : 2017.3.1

I have dotNet 4.6.2 installed on my machine, I don't have much clue about what is happening here.

![unity_2018-03-22_17-12-29](https://user-images.githubusercontent.com/2767681/37782858-45f5eb5e-2df4-11e8-86e0-b75fd8729b1b.png)

**Comment by IvanPashchenko** [comment]
@RDeluxe could you please look at what version of MSBuild is being used in Rider for your project?
Error message looks like you are using the some old version.

The settings can be found here: Preferences | Build, Execution, Deployment | Toolset and Build

**Comment by van800** [comment]
https://rider-support.jetbrains.com/hc/en-us/articles/207288089-Using-Project-Rider-under-Windows-without-Visual-Studio-prerequisites

**Comment by van800** [comment]
Feel free to ask here or reopen issue, if the issue is not resolved.

**Comment by RDeluxe** [comment]
Seems that (nuget tool is necessary)[https://stackoverflow.com/questions/47797510/the-getreferencenearesttargetframeworktask-task-was-not-found]. I'll post a comment on the previous link ! Thanks again

**Comment by van800** [comment]
You may try our MSBuild https://rider-support.jetbrains.com/hc/en-us/articles/207288089?page=1#comment_360000029404

**Comment by van800** [comment]
MS claim this is fixed, but not released yet
https://developercommunity.visualstudio.com/content/problem/137779/the-getreferencenearesttargetframeworktask-task-wa.html

**Comment by RDeluxe** [comment]
FIY : I tried uninstalling all the MS tools and using only your MSBuild, without luck (same error than above).

**Comment by van800** [comment]
Thanks for reporting, I will check our MSBuild. 
Meanwhile you may use either MSbuild 2017 from MS + nuget components or [MS Build 2015](https://www.microsoft.com/en-us/download/details.aspx?id=48159).
(You will need to select it in Rider Settings after installation)

**Comment by RDeluxe** [comment]
Yep, I felt back to MSBuild 2017 + nuget component from the MSBuild installer and it's working.

**Comment by van800** [comment]
I have reproduced the issue with our MSBuild https://youtrack.jetbrains.com/issue/RIDER-14558
So I am closing this. Thanks again for reporting.

**Comment by oestrada1001** [comment]
After creating a project in Visual Studio and opening it with JetBrain Rider, I stumbled across this issue. I had a pre-existing Xamarin project that wouldn't bundle after opening the new project. The default detected MSBuild Rider picked up; it wouldn't work.

I was able to fix it by using MSBuild in Visual Studio. I'm using a mac, so this was the path:
`/Applications/Visual Studio.app/Contents/MonoBundle/MSBuild/Current/bin/MSBuild.dll`
